---
title: "T-ALL Project: Download"
author: "HoangAnhNguyen"
date: "2023-04-03"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r install and load TCGAbiolinks}
#install.packages("BiocManager")
#BiocManager::install("TCGAbiolinks")

library(TCGAbiolinks)
```

# Please update the following line to suit your own needs
```{r set your own data directory}
GDCdata.dir <- "/home/tichit/sync/Recherche/AnnieNguyen/networkTALL/GDCdata"
```

```{r}
#STEP 0 

#Fetch data T-ALL P2
#mRNA P2
query_mRNA_P2 <- GDCquery(
    project = "TARGET-ALL-P2",
    data.category = "Transcriptome Profiling",
    data.type = "Gene Expression Quantification"
  )
#SNV P2 
query_SNV_P2 <- GDCquery(
    project = "TARGET-ALL-P2",
    data.category = "Simple Nucleotide Variation",
    access = "open",
    data.type = "Masked Somatic Mutation"
  )
#T-ALL P2 Clinical Data 
query_P2_clinical <- GDCquery(
  project = "TARGET-ALL-P2",
  data.category = "Clinical",
  access = "controlled",
  data.type = "Clinical Supplement"
  )
#Fetch data T-ALL P3
#mRNA P3
query_mRNA_P3 <- GDCquery(
    project = "TARGET-ALL-P3",
    data.category = "Transcriptome Profiling",
    data.type = "Gene Expression Quantification"
  )
#SNV P3 
query_SNV_P3 <- GDCquery(
    project = "TARGET-ALL-P3",
    data.category = "Simple Nucleotide Variation",
    access = "open",
    data.type = "Masked Somatic Mutation"
  )
#P3 T-ALL Clincial Data 
query_P3_clinical <- GDCquery(
  project = "TARGET-ALL-P3",
  data.category = "Clinical",
  access = "open",
  data.type = "Clinical Supplement"
  )
#Fetch TARGET-AML
#mRNA AML
query_mRNA_AML <- GDCquery(
    project = "TARGET-AML",
    data.category = "Transcriptome Profiling",
    data.type = "Gene Expression Quantification"
  )
#SNV AML
query_SNV_AML <- GDCquery(
    project = "TARGET-AML",
    data.category = "Simple Nucleotide Variation",
    access = "open",
    data.type = "Masked Somatic Mutation"
  )
#AML Clinical Data 
query_AML_clinical <- GDCquery(
  project = "TARGET-AML",
  data.category = "Clinical",
  access = "open",
  data.type = "Clinical Supplement"
  )

GDCdownload(query = query_mRNA_P2, directory = GDCdata.dir)
GDCdownload(query = query_SNV_P2, directory = GDCdata.dir)
GDCdownload(query = query_P2_clinical, directory = GDCdata.dir)
GDCdownload(query = query_mRNA_P3, directory = GDCdata.dir)
GDCdownload(query = query_SNV_P3, directory = GDCdata.dir)
GDCdownload(query = query_P3_clinical, directory = GDCdata.dir)
GDCdownload(query = query_mRNA_AML, directory = GDCdata.dir)
GDCdownload(query = query_SNV_AML, directory = GDCdata.dir)
GDCdownload(query = query_AML_clinical, directory = GDCdata.dir)
```

## TODO add code to fetch list of samples corresponding to ambiguous L
## currently it seems the only possibility is to go to GDC portal and download manually
## https://portal.gdc.cancer.gov/exploration?facetTab=clinical&filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.diagnoses.primary_diagnosis%22%2C%22value%22%3A%5B%22mixed%20phenotype%20acute%20leukemia%2C%20b%2Fmyeloid%2C%20nos%22%2C%22mixed%20phenotype%20acute%20leukemia%2C%20t%2Fmyeloid%2C%20nos%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.program.name%22%2C%22value%22%3A%5B%22TARGET%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.project_id%22%2C%22value%22%3A%5B%22TARGET-ALL-P3%22%5D%7D%7D%2C%7B%22content%22%3A%7B%22field%22%3A%22genes.is_cancer_gene_census%22%2C%22value%22%3A%5B%22true%22%5D%7D%2C%22op%22%3A%22in%22%7D%5D%7D
