---
title: "t-ALL_project_internshipM1"
author: "HoangAnhNguyen"
date: "2023-04-03"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r install and load packages}
#install.packages("BiocManager")
#BiocManager::install("DESeq2")

```

```{r}
Project.dir <- "/home/tichit/sync/Recherche/AnnieNguyen/networkTALL/"
GDCdata.dir <- paste0(Project.dir,"GDCdata/")
```

```{r build df from data directory, message = FALSE}
# library(TCGAbiolinks)
# 
# query_mRNA_P2 <- GDCquery(
#     project = "TARGET-ALL-P2",
#     data.category = "Transcriptome Profiling",
#     data.type = "Gene Expression Quantification"
#   )
# #SNV P2 
# query_SNV_P2 <- GDCquery(
#     project = "TARGET-ALL-P2",
#     data.category = "Simple Nucleotide Variation",
#     access = "open",
#     data.type = "Masked Somatic Mutation"
#   )
# #T-ALL P2 Clinical Data 
# query_P2_clinical <- GDCquery(
#   project = "TARGET-ALL-P2",
#   data.category = "Clinical",
#   access = "controlled",
#   data.type = "Clinical Supplement"
#   )
# #Fetch data T-ALL P3
# #mRNA P3
# query_mRNA_P3 <- GDCquery(
#     project = "TARGET-ALL-P3",
#     data.category = "Transcriptome Profiling",
#     data.type = "Gene Expression Quantification"
#   )
# #SNV P3 
# query_SNV_P3 <- GDCquery(
#     project = "TARGET-ALL-P3",
#     data.category = "Simple Nucleotide Variation",
#     access = "open",
#     data.type = "Masked Somatic Mutation"
#   )
# #P3 T-ALL Clincial Data 
# query_P3_clinical <- GDCquery(
#   project = "TARGET-ALL-P3",
#   data.category = "Clinical",
#   access = "open",
#   data.type = "Clinical Supplement"
#   )
# #Fetch TARGET-AML
# #mRNA AML
# query_mRNA_AML <- GDCquery(
#     project = "TARGET-AML",
#     data.category = "Transcriptome Profiling",
#     data.type = "Gene Expression Quantification"
#   )
# #SNV AML
# query_SNV_AML <- GDCquery(
#     project = "TARGET-AML",
#     data.category = "Simple Nucleotide Variation",
#     access = "open",
#     data.type = "Masked Somatic Mutation"
#   )
# #AML Clinical Data 
# query_AML_clinical <- GDCquery(
#   project = "TARGET-AML",
#   data.category = "Clinical",
#   access = "open",
#   data.type = "Clinical Supplement"
#   )
# 
# expdat_mRNA_P2 <- GDCprepare(query = query_mRNA_P2, directory = GDCdata.dir)
# expdat_SNV_P2 <- GDCprepare(query = query_SNV_P2, directory = GDCdata.dir)
# expdat_P2_clinical <- GDCprepare(query = query_P2_clinical, directory = GDCdata.dir)
# expdat_mRNA_P3 <- GDCprepare(query = query_mRNA_P3, directory = GDCdata.dir)
# expdat_SNV_P3 <- GDCprepare(query = query_SNV_P3, directory = GDCdata.dir)
# expdat_P3_clinical <- GDCprepare(query = query_P3_clinical, directory = GDCdata.dir)
# expdat_mRNA_AML <- GDCprepare(query = query_mRNA_AML, directory = GDCdata.dir)
# expdat_SNV_AML <- GDCprepare(query = query_SNV_AML, directory = GDCdata.dir)
# expdat_AML_clinical <- GDCprepare(query = query_AML_clinical, directory = GDCdata.dir)
```

```{bash, engine.opts='-l'}
PROJECT_DIR="/home/tichit/sync/Recherche/AnnieNguyen/networkTALL/"
PROCESSEDDATA_DIR="$PROJECT_DIR/processed_data/"

for I in P2 P3
do
  MSM_DIR="$PROJECT_DIR/GDCdata/TARGET-ALL-${I}/harmonized/Simple_Nucleotide_Variation/Masked_Somatic_Mutation/"
  zcat $GDCDATA_DIR/$MSM_DIR/*/*.gz | grep -v '^#' | grep -v '^Hugo' > "$PROCESSEDDATA_DIR/tALL_SNV_${I}.tsv"
done
```

```{r}
#retrieve the concat files of tALL SNV P2 + P3

#concat file TALL_SNV_P3

tAll_P3_SNV_all <- read.table(file = paste0(Project.dir,'processed_data/tALL_SNV_P3.tsv'), header = FALSE, quote='"', sep="\t")

#concat file TALL_SNv_P2

tALL_P2_SNV_all <- read.table(file = paste0(Project.dir,'processed_data/tALL_SNV_P2.tsv'), header = FALSE, quote='"', sep = "\t")

```




```{r}
#DATA PRESENTATION for T-ALL [MATRIX + PLOT]
library(tidyr)
library(dplyr)

P2_clinical_data <- read.table(file = paste0(GDCdata.dir, "TARGET-ALL-P2/harmonized/Clinical/TARGET_ALL_classification.txt"), sep = "\t")
colnames(P2_clinical_data) <- c("patientID", "disease")


matrix_P2_SNV <- matrix(c(tALL_P2_SNV_all$V1, tALL_P2_SNV_all$V16), ncol =  2)
df_P2_SNV <- as.data.frame(matrix_P2_SNV) %>% separate(V2, into = c("patientID","day"), sep = 16)
colnames(df_P2_SNV) <- c("geneID", "patientID", "day")
df_P2_SNV <- df_P2_SNV[, 1:2]
P2_join <- left_join(df_P2_SNV, P2_clinical_data, by ="patientID")

P2_TALL <- na.omit(P2_join[P2_join$disease == "T Cell ALL", ])

#number of patient carrying a mutation per gene 
P2_TALL_mut_per_gene <- P2_TALL %>%
  group_by(P2_TALL$geneID) %>%
  summarize(n = n())

P2_TALL_mut_per_gene <- P2_TALL_mut_per_gene[order(P2_TALL_mut_per_gene$n, decreasing = TRUE),]
P2_TALL_mut_per_gene <- as.data.frame(P2_TALL_mut_per_gene)

colnames(P2_TALL_mut_per_gene) <- c("Gene","N_mutation")


#Plot the most mutated gene in T-ALL leukemia 
P2_plot_gene_TALL <- ggplot(data = head(P2_TALL_mut_per_gene), aes(x = Gene, y = N_mutation)) + geom_bar(stat= "identity")
P2_plot_gene_TALL + coord_flip()

```



```{r}
#DATA PRESENTATION for B-ALL [MATRIX + PLOT]
library(ggplot2)
library(readxl)

matrix_P2_SNV_Ball <- matrix(c(tALL_P2_SNV_all$V1, tALL_P2_SNV_all$V16), ncol =  2)
df_P2_SNV_Ball <- as.data.frame(matrix_P2_SNV)%>% separate(V2,into = c("patientID","day"),sep = 16)
colnames(df_P2_SNV_Ball) <- c("geneID","patientID","day")
df_P2_SNV_Ball <- df_P2_SNV_Ball[,1:2]
P2_join_Ball <- left_join(df_P2_SNV_Ball,P2_clinical_data, by ="patientID")

P2_BALL <- na.omit(P2_join[P2_join$disease == "B Cell ALL",])

#number of patient carrying a mutation per gene 
P2_BALL_mut_per_gene <- P2_BALL %>%
  group_by(P2_BALL$geneID) %>%
  summarize(n=n())

P2_BALL_mut_per_gene <- P2_BALL_mut_per_gene[order(P2_BALL_mut_per_gene$n,decreasing = TRUE),]
P2_BALL_mut_per_gene <- as.data.frame(P2_BALL_mut_per_gene)

colnames(P2_BALL_mut_per_gene) <- c("Gene","N_mutation")


#Plot the most mutated gene in T-ALL leukemia 
P2_plot_gene_BALL <- ggplot(data = head(P2_BALL_mut_per_gene), aes(x = Gene, y = N_mutation)) + geom_bar(stat= "identity")
P2_plot_gene_BALL + coord_flip()

```



```{r}
#DATA PRESENTATION for AML [MATRIX + PLOT]
library(ggplot2)
library(readxl)

matrix_P3_SNV_AML <- matrix(c(tAll_P3_SNV_all$V1, tAll_P3_SNV_all$V16), ncol =  2)
df_P3_SNV_AML <- as.data.frame(matrix_P3_SNV_AML)%>% separate(V2,into = c("patientID","day"),sep = 22)
colnames(df_P3_SNV_AML) <- c("geneID","patientID","day")
df_P3_SNV_AML <- df_P3_SNV_AML[,1:2]

P3_bALL_clinical_data <- read_xlsx("C:/Users/nguye/OneDrive/Bureau/internshipDLAD_M1/GDCdata/TARGET-ALL-P3/harmonized/Clinical/Clinical_Supplement/b524f6f8-32dd-4edc-abbf-ae55fe533a67/TARGET_ALL_ClinicalData_Phase_III_20210811.xlsx")
P3_bALL_clinical_data <- data.frame(geneID = P3_bALL_clinical_data$`TARGET USI`, patientID = P3_bALL_clinical_data$...)

P3_join_AML <- left_join(df_P3_SNV_AML,P3_bALL_clinical_data, by ="patientID")

P3_AML <- na.omit(P2_join[P2_join$disease == "AML",])

#number of patient carrying a mutation per gene 
P2_BALL_mut_per_gene <- P2_BALL %>%
  group_by(P2_BALL$geneID) %>%
  summarize(n=n())

P2_BALL_mut_per_gene <- P2_BALL_mut_per_gene[order(P2_BALL_mut_per_gene$n,decreasing = TRUE),]
P2_BALL_mut_per_gene <- as.data.frame(P2_BALL_mut_per_gene)

colnames(P2_BALL_mut_per_gene) <- c("Gene","N_mutation")


#Plot the most mutated gene in T-ALL leukemia 
P2_plot_gene_BALL <- ggplot(data = head(P2_BALL_mut_per_gene), aes(x = Gene, y = N_mutation)) + geom_bar(stat= "identity")
P2_plot_gene_BALL + coord_flip()

```

```{r}
#Call clinical Data P2
P2_clinical_data <- read.table(file = "C:/Users/nguye/OneDrive/Bureau/internshipDLAD_M1/GDCdata/TARGET-ALL-P2/harmonized/Clinical/TARGET_ALL_classification.txt", sep = "\t")
colnames(P2_clinical_data) <- c("patientID","disease")


plot_P2_SNV <- ggplot(matrix_P2_SNV, aes(x = matrix_P2_SNV[,2], y = matrix_P2_SNV[,1], color = matrix_P2_SNV[,3]))+ labs(x = "Sample", y = "Gene", z = "SNV types", tittle ="T-ALL SNV P2")  + 
  geom_jitter(width = 0.5, height = 0.5)

plot_P2_SNV

heatmap(matrix_P2_SNV, scale = "none")
```



################################################################################################################################################################################
#########################################################################################################################
```{r}

#retrieve the concat files of tALL SNV P2 + P3

#concat file TALL_SNV_P3

tAll_P3_SNV_all <- read.table(file = 'C:/Users/nguye/OneDrive/Bureau/internshipDLAD_M1/GDCdata/TARGET-ALL-P3/harmonized/Simple_Nucleotide_Variation/Masked_Somatic_Mutation/tALL_SNV_P3.tsv', header = FALSE,sep="\t")

#concat file TALL_SNv_P2

tAll_P2_SNV_all <- read.table(file = 'C:/Users/nguye/OneDrive/Bureau/internshipDLAD_M1/GDCdata/TARGET-ALL-P2/harmonized/Simple_Nucleotide_Variation/Masked_Somatic_Mutation/tALL_SNV_P2.tsv', header = FALSE, sep = "\t")

```


```{r}
A <- tAll_P2_SNV_all %>%
  group_by(tAll_P2_SNV_all$V16) %>%
  summarize(n=n())


B <- tAll_P3_SNV_all %>%
  group_by(tAll_P3_SNV_all$V16) %>%
  summarize(n=n())

boxplot(A$n, ylim= c(0,50))
boxplot(B$n, ylim= c(0,50))

plot_P2_SNV_1 <- ggplot(A, aes(x = A$`tAll_P2_SNV_all$V16`, y = A$n) + labs(x = "Sample", y = "Gene")) + geom_jitter(width = 0.5, height = 0.5)

plot_P2_SNV_1

```








#########################################################################################################################
```{r}
#Determine number of mutation occurrence per gene per patient 

mut_per_gene_P3 <- tAll_P3_SNV_all  %>% group_by(tAll_P3_SNV_all$V1) %>% summarise(total_count=n()) mut_per_gene_P3
 
most_common_gene_P3 <-names(which.max(table(mut_per_gene_P3$`tAll_P3_SNV_all$V1`)))
most_common_gene_P3



mut_per_gene_P2 <- tAll_P2_SNV_all  %>% group_by(tAll_P2_SNV_all$V1) %>% summarise(total_count=n())
mut_per_gene_P2 

most_common_gene_P2 <-names(which.max(table(mut_per_gene_P2$`tAll_P2_SNV_all$V1`)))
most_common_gene_P2
```



```{r}
#Clinical Data in Target_ALL_P2 

library(readxl)

P2_Target_All_clindata_valsup <- read_xlsx("C:/Users/nguye/OneDrive/Bureau/internshipDLAD_M1/GDCdata/TARGET-ALL-P2/harmonized/Clinical/TARGET_ALL_ClinicalData_Validation_Supplement_20190129.xlsx")
Target_All_clindata_valsup


#Clinical Data in Target_ALL_P3
P3_Target_All_clindata_valsup <- read_xlsx("C:/Users/nguye/OneDrive/Bureau/internshipDLAD_M1/GDCdata/TARGET-ALL-P3/harmonized/Clinical/Clinical_Supplement/b524f6f8-32dd-4edc-abbf-ae55fe533a67/TARGET_ALL_ClinicalData_Phase_III_20210811.xlsx")
P3_Target_All_clindata_valsup
```















```{r}


#########################################################################################################################
########DIFFERENTIAL EXPRESSION USING LIMMA                                     #########################################
#########################################################################################################################
#STEP1 - Differential Expression in Target_ALL_P2 Using LIMMA 


#TEST by ANNIE#

#0-Access data (group all data in mRNA expression together)
all_files <- (Sys.glob(paths = "C:/Users/nguye/OneDrive/Bureau/internshipDLAD_M1/GDCdata/TARGET-ALL-P2/harmonized/Transcriptome_Profiling/Gene_Expression_Quantification/*/*.tsv"))

listOfFiles <- lapply(all_files, function(x) read.table(x, header = TRUE, sep = "\t")) 
listOfFiles

exp_P2_test1 <- read.table(file = "C:/Users/nguye/OneDrive/Bureau/internshipDLAD_M1/GDCdata/TARGET-ALL-P2/harmonized/Transcriptome_Profiling/Gene_Expression_Quantification/0a63c7cf-e08e-4ef6-87b0-6d2b2adf928f/d918c0aa-47a1-4fda-b4c7-560929fdb927.rna_seq.augmented_star_gene_counts.tsv", header = TRUE , sep = "\t")
exp_P2_test1

#1-Fit the model
fit_exp_P2_test1 <- lmFit(exp_P2_test1$fpkm_unstranded)
fit_exp_P2_test1

#2-Calculate the t-statistics
fit_exp_P2_test1_tstat <- eBayes(fit_exp_P2_test1) 
fit_exp_P2_test1_tstat

#3-Summarize the result 
result_exp_P2_test1 <- decideTests(fit_exp_P2_test1_tstat$F.p.value)
result_exp_P2_test1
summary(result_exp_P2_test1)
```


```{r}
############################################################################
###########   Annie's attempt to extract data from expression data      #### 
############################################################################

#####Pull the name of patient(cases) in this study####### 
P2_mRNA_cases <- query_mRNA_P2[[1]][[1]]$cases

#####"get list of interesting files
paths <- list.files(
    path = 'C:/Users/nguye/OneDrive/Bureau/internshipDLAD_M1/GDCdata/TARGET-ALL-P2/harmonized/Transcriptome_Profiling/Gene_Expression_Quantification/',
    full.name = TRUE,
    recursive = TRUE,
    pattern = '*.tsv'
  )
#their number
nb.samples <- length(paths)
#the short name of each file: sample ID
sample.names <- substr(basename(paths), 1, 36)

sample.names <- P2_mRNA_cases

#get transcript names from first file: paths[1]
#skip the first 6 rows (uninteresting)
transcript.names <- read.delim(paths[1], skip = 6, header = FALSE)[1][[1]] #get only the 1st column [1] then extract to vector [[1]]
gene.names <- read.delim(paths[1], skip = 6, header = FALSE)[2][[1]] #get only the 1st column [1] then extract to vector [[1]]

#their number  
nb.transcripts <- length(transcript.names)
nb.genes <- length(gene.names)
nb.samples <- length(sample.names)
#create empty matrix of right size
m <- matrix(
    nrow = nb.genes,
    ncol = nb.samples,
    byrow = TRUE,
    dimnames = list(gene.names,sample.names)
  )

#Fill the empty matrix with data#

for (i in 1:length(paths)) {
  print(paste0("processing file ", i, "/", nb.samples, ": ", basename(paths[[i]])))
  m[,i]  <- read.delim(paths[[i]], skip = 6, header = FALSE)[4][[1]]
  #m[,i]  <- read.delim(paths[[i]], skip = 6, header = FALSE)[4][[1]]
}


```


```{r}
##########################################################
#concatenate thousand of files by columns---NOT WORKING ##
##########################################################

# get a list of files

my.file.list <- list.files(
    path = 'C:/Users/nguye/OneDrive/Bureau/internshipDLAD_M1/GDCdata/TARGET-ALL-P2/harmonized/Transcriptome_Profiling/Gene_Expression_Quantification/',
    full.name = TRUE,
    recursive = TRUE,
    pattern = '*.tsv'
  )
# for each file, run read.table and select only the first column
my.list <- lapply(X = my.file.list, FUN = function(x) {
            read.delim(my.file.list[1], skip = 6, header = FALSE)[4][[1]]
        })
# merge columns that are in a list into one data.frame
my.df <- do.call("cbind", my.list)

##############################################################################





```



```{r}
####################################################################################
#######     SUGGESTION BY LAURENT to extract expression data        ################
####################################################################################

#get list of interesting files
paths <- list.files(
    path = 'C:/Users/nguye/OneDrive/Bureau/internshipDLAD_M1/GDCdata/TARGET-ALL-P2/harmonized/Transcriptome_Profiling/Gene_Expression_Quantification/',
    full.name = TRUE,
    recursive = TRUE,
    pattern = '*.tsv'
  )
#their number
nb.samples <- length(paths)
#the short name of each file: sample ID
sample.names <- substr(basename(paths), 1, 36)

sample.names <- P2_mRNA_cases

#get transcript names from first file: paths[1]
#skip the first 6 rows (uninteresting)
transcript.names <- read.delim(paths[1], skip = 6, header = FALSE)[1][[1]] #get only the 1st column [1] then extract to vector [[1]]
gene.names <- read.delim(paths[1], skip = 6, header = FALSE)[2][[1]] #get only the 1st column [1] then extract to vector [[1]]

#their number
nb.transcripts <- length(transcript.names)

#create empty matrix of right size
m <- matrix(
    nrow = nb.samples,
    ncol = nb.transcripts,
    byrow = TRUE,
    dimnames = list(sample.names, transcript.names)
  )

# fill matrix row by row (sample by sample)
# we use the 4th row only (unstranded raw count)

for (i in 1:length(paths)) {
  print(paste0("processing file ", i, "/", nb.samples, ": ", basename(paths[[i]])))
  m[i]  <- read.delim(paths[[i]], skip = 6, header = FALSE)[4][[1]]
}

write.table(m, "RNASeq_P2.tsv")
```






#########################################################################################################################
#########################################################################################################################
```{r}
#STEP1 - Differential Expression in Target_ALL_P2 Using DeSeq2

library(DESeq2)

#Step 1.1 Normalize Count


```



##################################################################################################################################################################################################################################################
```{r}
#DownLoad data Storage 

saveRDS(tAll_P2_SNV_all, file = "C:/Users/nguye/OneDrive/Bureau/internshipDLAD_M1/R_scripts/DownloadedData/tALL_P2_SNV_all.rds")
saveRDS(tAll_P3_SNV_all, file = "C:/Users/nguye/OneDrive/Bureau/internshipDLAD_M1/R_scripts/DownloadedData/tALL_P3_SNV_all.rds")
  
    
```


